xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/usersandgroups/usersandgroups.js

```diff
@@ -142,102 +147,102 @@ window.MSCheckbox = Class.create({
 
       var action = "";
       var nxtst = (self.state + 1) % self.nrstates;
-
-      // 1. The current user is clearing / denying himself any right.
-      if (self.currentUorG == window.currentUser) {
-        if (nxtst == 2) {
-          var denymessage = "$escapetool.javascript($services.localization.render('rightsmanager.denyrightforcurrentuser'))".replace('__right__', self.right);
-          if (!confirm(denymessage)) {
-            var clearmessage = "$escapetool.javascript($services.localization.render('rightsmanager.clearrightforcurrentuserinstead'))".replace('__right__', self.right);
-            if (confirm(clearmessage)) {
-              action = "clear";
-              self.state = 2;
-              nxtst = 0;
-            } else {
+      require(['xwiki-l10n!users-and-groups-translation-keys'], function(l10n) {
```

Comment:
I'm still not fully convinced by this part being asynchronous. Couldn't we define this at a more global place. cc @mflorea if you have some idea.

Reply:
I tried to require the translations above the MSCheckbox class definition, but this ended in an error in another (synchronous) script that couldn't find MSCheckbox.